src/core/app_irc_event.c

```diff
@@ -43,88 +43,88 @@
 
 static gboolean do_period_ping(gpointer user_data);
 static void add_numeric_error_message(SrnChat *chat, int event, const char
-        *origin, const char **params, int count, const SircMessageContext *context);
+        *origin, const char **params, int count, SircMessageContext *context);
 static void rejoin_all_channels(SrnServer *srv);
 static gboolean rejoin_all_channels_cb(gpointer user_data);
 
 static void irc_event_connect(SircSession *sirc, const char *event,
-        const SircMessageContext *context);
```

Comment:
Why we need to remove the const modifier?

Reply:
It's not possible to use `g_date_time_ref` on a const object, so `sirc_message_context_get_time` needs to return a non-const `GDateTime`, so it needs to be passed a non-const `SircMessageContext`, which propagates through most of the code. So I just removed the `const` qualifier everywhere.

Reply:
I don't think so, `const` only limits you to change the directly content of address (for example, you can't rewrite the `time` field of an `const *SircMessageContext`), but change the content of `time` field is possible. 

See #357.